pkg/resource/health_check/sdk.go

```diff
@@ -0,0 +1,893 @@
+			f2.ResourcePath = resp.HealthCheck.HealthCheckConfig.ResourcePath
+		}
+		if resp.HealthCheck.HealthCheckConfig.RoutingControlArn != nil {
+			f2.RoutingControlARN = resp.HealthCheck.HealthCheckConfig.RoutingControlArn
+		}
+		if resp.HealthCheck.HealthCheckConfig.SearchString != nil {
+			f2.SearchString = resp.HealthCheck.HealthCheckConfig.SearchString
+		}
+		if resp.HealthCheck.HealthCheckConfig.Type != nil {
+			f2.Type = resp.HealthCheck.HealthCheckConfig.Type
+		}
+		ko.Spec.HealthCheckConfig = f2
+	} else {
+		ko.Spec.HealthCheckConfig = nil
+	}
+	if resp.HealthCheck.HealthCheckVersion != nil {
+		ko.Status.HealthCheckVersion = resp.HealthCheck.HealthCheckVersion
+	} else {
+		ko.Status.HealthCheckVersion = nil
+	}
+	if resp.HealthCheck.Id != nil {
+		ko.Status.ID = resp.HealthCheck.Id
+	} else {
+		ko.Status.ID = nil
+	}
+	if resp.HealthCheck.LinkedService != nil {
+		f5 := &svcapitypes.LinkedService{}
+		if resp.HealthCheck.LinkedService.Description != nil {
+			f5.Description = resp.HealthCheck.LinkedService.Description
+		}
+		if resp.HealthCheck.LinkedService.ServicePrincipal != nil {
+			f5.ServicePrincipal = resp.HealthCheck.LinkedService.ServicePrincipal
+		}
+		ko.Status.LinkedService = f5
+	} else {
+		ko.Status.LinkedService = nil
+	}
+
+	rm.setStatusDefaults(ko)
+	return &resource{ko}, nil
+}
+
+// requiredFieldsMissingFromReadOneInput returns true if there are any fields
+// for the ReadOne Input shape that are required but not present in the
+// resource's Spec or Status
+func (rm *resourceManager) requiredFieldsMissingFromReadOneInput(
+	r *resource,
+) bool {
+	return r.ko.Status.ID == nil
+
+}
+
+// newDescribeRequestPayload returns SDK-specific struct for the HTTP request
+// payload of the Describe API call for the resource
+func (rm *resourceManager) newDescribeRequestPayload(
+	r *resource,
+) (*svcsdk.GetHealthCheckInput, error) {
+	res := &svcsdk.GetHealthCheckInput{}
+
+	if r.ko.Status.ID != nil {
+		res.SetHealthCheckId(*r.ko.Status.ID)
+	}
+
+	return res, nil
+}
+
+// sdkCreate creates the supplied resource in the backend AWS service API and
+// returns a copy of the resource with resource fields (in both Spec and
+// Status) filled in with values from the CREATE API operation's Output shape.
+func (rm *resourceManager) sdkCreate(
+	ctx context.Context,
+	desired *resource,
+) (created *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkCreate")
+	defer func() {
+		exit(err)
+	}()
+	input, err := rm.newCreateRequestPayload(ctx, desired)
+	if err != nil {
+		return nil, err
+	}
+
+	// You must use a unique CallerReference string every time you submit a
+	// CreateHealthCheck request. CallerReference can be any unique string, for
+	// example, a date/timestamp.
+	// TODO: Name is not sufficient, since a failed request cannot be retried.
+	// We might need to import the `time` package into `sdk.go`
+	input.SetCallerReference(getCallerReference())
```

Comment:
Since this file is fully generated, this block of code needs to be injected via `generator.yaml` hooks directives

Reply:
Ah right, I did not know how those work, so yes I will move it.